### ISSUE TITLE
how to get metrics from C_AP

### ISSUE THREAD

**Issue by osgrass** [microsoft/LightGBM#3103]
Hi, I have a question regarding how to obtain metrics of training/validation from each iteration. I am calling c_api from c++ library. I see this in the document, I wonder if it can be done from c_api? Any clues would be appreciated. Thanks.

Xinmin

is_provide_training_metric 🔗︎, default = false, type = bool, aliases: training_metric, is_training_metric, train_metric

set this to true to output metric result over training dataset

Note: can be used only in CLI version

## How you are using LightGBM?

LightGBM component: 

## Environment info

Operating System:

CPU/GPU model:

C++ compiler version:

CMake version:

Java version:

Python version:

R version:

Other:

LightGBM version or commit hash:

## Steps to reproduce

1.
2.
3.

**Comment by guolinke** [comment]
you can refer our python-package code:
https://github.com/microsoft/LightGBM/blob/7fe10fa10b529805e447de523d56bbf8a589ae23/python-package/lightgbm/basic.py#L2880-L2911

basically, you need to call LGBM_BoosterGetEval.
